### PR TITLE
[Utility][Symbols] Do not require a SymbolServerType when only "Index Sources" is checked.

### PR DESCRIPTION
**Pull request by jlaanstra**
This PR update the PublishSymbols.ps1 to only require the 'SymbolServerType' parameter when 'PublishSymbols' is set to true.

To achieve this, line 37-68 have moved down to after the indexing step and the check on line 74 has been simplified to not require the 'SymbolServerType' parameter either as its value isn't used inside the conditional block anyway.

### CODE REVIEW

**Comment by jlaanstra** [comment]
Label: Utility

**Comment by jlaanstra** [comment]
@jerickmsft  @ericsciple based on previous PRs for symbols.

**Comment by mitchdenny** [comment]
Adding @arunkm

File: Tasks/PublishSymbols/PublishSymbols.ps1 [hunk 1]
```diff
@@ -31,47 +31,15 @@ try {
...
-        [string]$SymbolsVersion = Get-VstsInput -Name 'SymbolsVersion' -Default (Get-VstsTaskVariable -Name 'Build.BuildNumber' -Require)
-        [string]$SymbolsArtifactName = Get-VstsInput -Name 'SymbolsArtifactName'
-    }
     [bool]$SkipIndexing = -not (Get-VstsInput -Name 'IndexSources' -AsBool)
+    [bool]$NeedsPublishSymbols = Get-VstsInput -Name 'PublishSymbols' -Require -AsBool
     [bool]$TreatNotIndexedAsWarning = Get-VstsInput -Name 'TreatNotIndexedAsWarning' -AsBool
     [string]$defaultSymbolFolder = (Get-VstsTaskVariable -Name 'Build.SourcesDirectory' -Default "")
     [string]$SymbolsFolder = Get-VstsInput -Name 'SymbolsFolder' -Default $defaultSymbolFolder
-
-    if ( ($SymbolServerType -eq "FileShare") -or ($SymbolServerType -eq "TeamServices") -or (-not $SkipIndexing) ) {
```
Comment (arunkm) [inline Tasks/PublishSymbols/PublishSymbols.ps1]: >($SymbolServerType -eq "FileShare") -or ($SymbolServerType -eq "TeamServices") [](start = 9, length = 78)

The $fileList is used below to publish file, 
and will cause a regression ($fileList will be empty) when $SkipIndexing = true and publishing

**Comment by stephenmichaelf** [comment]
Closing due to age, please reopen if needed.

**Pull request closed without merging**